### PR TITLE
[fix] add dispatch queue sync around variable calls

### PR DESCRIPTION
**Pull request by chantalchung**
- to prevent crashes on multiple threads

### CODE REVIEW

File: DevCycle/DVCClient.swift [hunk 1]
```diff
@@ -252,8 +252,27 @@ public class DVCClient {
...
+        )
+        if (Thread.isMainThread) {
+            variable = variableInternal(key: key, defaultValue: defaultValue)
+        } else {
+            DispatchQueue.main.sync {
+                variable = variableInternal(key: key, defaultValue: defaultValue)
+            }
+        }
+        return variable
+    }
```
Comment (jonathannorris) [inline DevCycle/DVCClient.swift:273]: Can we simplify it to:

```suggestion
        if (Thread.isMainThread) {
            return variableInternal(key: key, defaultValue: defaultValue)
        } else {
            DispatchQueue.main.sync {
                return variableInternal(key: key, defaultValue: defaultValue)
            }
        }
    }
```

Comment (chantalchung) [reply DevCycle/DVCClient.swift:273]: that won't work as we have to ensure we return a value outside of the DispatchQueue
image

Comment (jonathannorris) [reply DevCycle/DVCClient.swift:273]: hum... does this work?

```suggestion
        if (Thread.isMainThread) {
            return variableInternal(key: key, defaultValue: defaultValue)
        } else {
            let variable
            DispatchQueue.main.sync {
                variable = variableInternal(key: key, defaultValue: defaultValue)
            }
            return variable
        }
    }
```

Comment (chantalchung) [reply DevCycle/DVCClient.swift:273]: `variable` still needs to be initialized or else there would be nothing to return at the end of the function, so the best we could do is move the default into the else?
```
    public func variable<T>(key: String, defaultValue: T) -> DVCVariable<T> {
        if (Thread.isMainThread) {
            return variableInternal(key: key, defaultValue: defaultValue)
        } else {
            var variable = DVCVariable(
                key: key,
                type: String(describing: T.self),
                value: nil,
                defaultValue: defaultValue,
                evalReason: nil
            )
            DispatchQueue.main.sync {
                variable = variableInternal(key: key, defaultValue: defaultValue)
            }
            return variable
        }
    }
```

Comment (chantalchung) [reply DevCycle/DVCClient.swift:273]: image

**Review by jonathannorris — APPROVED** [review]

**Pull request merged**